### PR TITLE
FRESH-506 #175 C_Invoice_Update_PaymentRule

### PR DESCRIPTION
**Pull request by metas-rc**
Process and schedule added for 23:30 each evening

### CODE REVIEW

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/ddl/functions/C_Invoice_Update_PaymentRule.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+-- FRESH-506
+-- This function changes the PaymentRule from 'S' (Check) to 'P' (OnCredit)
+CREATE OR REPLACE FUNCTION C_Invoice_Update_PaymentRule ()
+
+RETURNS void
+
+AS
+$$
+
+UPDATE
```
Comment (metas-ts) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/ddl/functions/C_Invoice_Update_PaymentRule.sql]: please also set Updated=now(), Updatedby=99

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+;
+
+-- 12.07.2016 17:14
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+DELETE FROM AD_Scheduler WHERE AD_Scheduler_ID=550037
+;
+
+-- 12.07.2016 17:14
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+UPDATE AD_Scheduler SET EntityType='de.metas.i',Updated=TO_TIMESTAMP('2016-07-12 17:14:28','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Scheduler_ID=550038
```
Comment (metas-ts) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql]: wrong entity type?

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+;
+
+-- 12.07.2016 17:04
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+INSERT INTO AD_Scheduler (AD_Client_ID,AD_Org_ID,AD_Role_ID,AD_Scheduler_ID,Created,CreatedBy,CronPattern,Description,EntityType,Frequency,FrequencyType,IsActive,IsIgnoreProcessingTime,KeepLogDays,ManageScheduler,Name,Processing,SchedulerProcessType,ScheduleType,Status,Supervisor_ID,Updated,UpdatedBy) VALUES (0,0,0,550037,TO_TIMESTAMP('2016-07-12 17:04:28','YYYY-MM-DD HH24:MI:SS'),100,'30 23 * * *','Task FRESH-506 : Change all invoices'' PaymentRUle values from ''S'' (Check) to ''P'' (OnCredit)','de.metas.s',1,'D','Y','N',7,'N','C_Invoice_Update_PaymentRule','N','P','C','NEW',0,TO_TIMESTAMP('2016-07-12 17:04:28','YYYY-MM-DD HH24:MI:SS'),100)
+;
+
+-- 12.07.2016 17:04
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+UPDATE AD_Scheduler SET EntityType='de.metas.i',Updated=TO_TIMESTAMP('2016-07-12 17:04:41','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Scheduler_ID=550037
```
Comment (metas-ts) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql]: wrong entity type? (also below)

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+;
+
+-- 12.07.2016 15:50
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+INSERT INTO AD_Process (AccessLevel,AD_Client_ID,AD_Org_ID,AD_Process_ID,AllowProcessReRun,Classname,CopyFromProcess,Created,CreatedBy,Description,EntityType,IsActive,IsApplySecuritySettings,IsBetaFunctionality,IsDirectPrint,IsOneInstanceOnly,IsReport,IsServerProcess,IsUseBPartnerLanguage,LockWaitTimeout,Name,RefreshAllAfterExecution,ShowHelp,SQLStatement,Statistic_Count,Statistic_Seconds,Type,Updated,UpdatedBy,Value) VALUES ('3',0,0,540701,'Y','de.metas.adempiere.process.ExecuteUpdateSQL','N',TO_TIMESTAMP('2016-07-12 15:50:25','YYYY-MM-DD HH24:MI:SS'),100,'Task FRESH-506 : Change all invoices'' PaymentRUle values from ''S'' (Check) to ''P'' (OnCredit)','de.metas.invoice','Y','N','N','N','N','N','N','Y',0,'C_Invoice_Update_PaymentRule','N','Y','INSERT INTO C_Invoice_Candidate_Recompute (C_Invoice_Candidate_ID)
```
Comment (metas-ts) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql]: which role does this process play?

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql [hunk 5]
```diff
@@ -0,0 +1,114 @@
+-- 12.07.2016 15:25
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+INSERT INTO AD_Process (AccessLevel,AD_Client_ID,AD_Org_ID,AD_Process_ID,AllowProcessReRun,Classname,CopyFromProcess,Created,CreatedBy,EntityType,IsActive,IsApplySecuritySettings,IsBetaFunctionality,IsDirectPrint,IsOneInstanceOnly,IsReport,IsServerProcess,IsUseBPartnerLanguage,LockWaitTimeout,Name,RefreshAllAfterExecution,ShowHelp,Statistic_Count,Statistic_Seconds,Type,Updated,UpdatedBy,Value) VALUES ('1',0,0,540700,'Y','de.metas.adempiere.process.ExecuteUpdateSQL','N',TO_TIMESTAMP('2016-07-12 15:25:44','YYYY-MM-DD HH24:MI:SS'),100,'U','Y','N','N','N','N','N','N','Y',0,'test','N','Y',0,0,'SQL',TO_TIMESTAMP('2016-07-12 15:25:44','YYYY-MM-DD HH24:MI:SS'),100,'testrux')
```
Comment (metas-ts) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql]: wrong entitytype 'U', I might be wrong but I think it's not updated further down

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+INSERT INTO AD_Process_Trl (AD_Language,AD_Process_ID, Description,Help,Name, IsTranslated,AD_Client_ID,AD_Org_ID,Created,Createdby,Updated,UpdatedBy) SELECT l.AD_Language,t.AD_Process_ID, t.Description,t.Help,t.Name, 'N',t.AD_Client_ID,t.AD_Org_ID,t.Created,t.Createdby,t.Updated,t.UpdatedBy FROM AD_Language l, AD_Process t WHERE l.IsActive='Y' AND l.IsSystemLanguage='Y' AND l.IsBaseLanguage='N' AND t.AD_Process_ID=540701 AND NOT EXISTS (SELECT * FROM AD_Process_Trl tt WHERE tt.AD_Language=l.AD_Language AND tt.AD_Process_ID=t.AD_Process_ID)
+;
+
+-- 12.07.2016 15:50
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+UPDATE AD_Process SET AccessLevel='7',Updated=TO_TIMESTAMP('2016-07-12 15:50:40','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Process_ID=540701
+;
+
+-- 12.07.2016 16:20
```
Comment (metas-ts) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql]: did the display logic cahnges work out? 
yes=> pls C&P the SQL to de.metas.adempiere.adempiere
no => pls delete those lines

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql (hunk 4 above)
Comment (metas-rc) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql]: Sorry. I tried something locally, about that SQLStatement I think, and I forgot to delete it.

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql (hunk 5 above)
Comment (metas-rc) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql]: Not needed, these lines were deleted. See next comment

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql (hunk 6 above)
Comment (metas-rc) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql]: Lines deleted. The logic doesn't work that's why I created this task: https://metasfresh.atlassian.net/browse/FRESH-520

File: de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql (hunk 2 above)
Comment (metas-rc) [inline de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/49-de.metas.invoice/5448170_sys_FRESH-506_Invoice_UpdatePaymentRule_Process_Schedule.sql]: fixed

**Pull request merged**